Firmware/mmu2.cpp

```diff
@@ -779,30 +779,36 @@ StepStatus MMU2::LogicStep() {
     case Processing:
         OnMMUProgressMsg(logic.Progress());
         break;
-    case CommandError:
-        ReportError(logic.Error(), ErrorSourceMMU);
-        break;
-    case CommunicationTimeout:
-        state = xState::Connecting;
-        ReportError(ErrorCode::MMU_NOT_RESPONDING, ErrorSourcePrinter);
-        break;
-    case ProtocolError:
-        state = xState::Connecting;
-        ReportError(ErrorCode::PROTOCOL_ERROR, ErrorSourcePrinter);
-        break;
-    case VersionMismatch:
-        StopKeepPowered();
-        ReportError(ErrorCode::VERSION_MISMATCH, ErrorSourcePrinter);
-        break;
     case ButtonPushed:
         lastButton = logic.Button();
         LogEchoEvent_P(PSTR("MMU Button pushed"));
         CheckUserInput(); // Process the button immediately
         break;
     default:
-        break;
+        if(reportErrors) {
+            switch (ss)
+            {
+            case CommandError:
+                ReportError(logic.Error(), ErrorSourceMMU);
+                break;
+            case CommunicationTimeout:
+                state = xState::Connecting;
```

Comment:
I don't feel comfortable with these CommunicationTimeout and ProtocolError states, but let's keep them for later.